### ISSUE TITLE
Email address has a wrong format error, on paypal express checkout Magento 2.3.2

### ISSUE THREAD

**Issue by Arsalanulhaq** [magento/magento2#27681]
### Preconditions (*)

1. Issue appears when pacing order with Paypal express checkout
2.

### Steps to reproduce (*)

1. Goto checkout and try order using Paypal express checkout
2.

### Expected result (*)

1. [Screenshots, logs or description]
2. Checkout should completed and order created in Magento

### Actual result (*)

1. [Screenshots, logs or description]
2. Checkout error `Exception message: We can't save the address:
Email has a wrong format
Trace: <pre>#1 Magento\Framework\Model\ResourceModel\Db\AbstractDb->save() called at [vendor/magento/framework/Model/AbstractModel.php:655]`

However payment is clearing and paypal email also received, but Magento order not created (Because of email validation error)
I checked in this file `vendor/magento/module-sales/Model/Order/Address/Validator.php` and $address->getEmail() is coming NULL/empty
https://i.stack.imgur.com/6yjfR.png $address->getEmail() is returning null, that's why validation is failing

**Comment by Arsalanulhaq** [comment]
Issue resolved with this patch https://nimb.ws/gFA06i
![screenshot-github com-2020 04 13-10_30_27](https://user-images.githubusercontent.com/26971013/79094763-f72e3b00-7d71-11ea-9cce-79c64a294a56.png)

**Comment by VigneshVenkatesan154** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @VigneshVenkatesan154. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @VigneshVenkatesan154, here is your [Magento instance](https://i-27681-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-27681-2-4-develop.instances.magento-community.engineering/admin_ae76](https://i-27681-2-4-develop.instances.magento-community.engineering/admin_ae76)
Login: `e3380317` Password: `3ff57c3052d6`
Instance will be terminated in up to 3 hours.

**Comment by VigneshVenkatesan154** [comment]
@Arsalanulhaq Can you please add more clear preconditions and Steps to reproduce

**Comment by rommelfreddy** [comment]
@VigneshVenkatesan154 just order with paypal express.
Got the same issue.

You will got this issue on the (magento) confirm page after clicking on the Checkout-Button.

**Comment by VigneshVenkatesan154** [comment]
@rommelfreddy @Arsalanulhaq   I tried with magento 2-4-instance. The issue is not recreated. Can you please give some insights like instance version etc. 
Please attached reference video:
https://drive.google.com/file/d/1wYsMdgrKh6dGPM69xa3QX90tTkWaeLID/view

**Comment by rommelfreddy** [comment]
@VigneshVenkatesan154 the video is very helpfull.
After the PayPal Popup you should be redirected to a other magento confirm page.
This page is just for the paypal express payments.

i am not sure how it works, but the paypal module differentiate between redirecting to the successful-page and the confim-page.

i haven't digged enough for that :/

**Comment by VigneshVenkatesan154** [comment]
Yes @rommelfreddy I do checked that scenario too. Basically the confirmation page will appear when we express checkout through paypal from cart or mini cart. It also worked as expected.

@Arsalanulhaq May be some insights from your end could please us reproduce the issue. A video would be helpful.

**Comment by Arsalanulhaq** [comment]
Yes, Issue appears when we click on paypal express checkout from product/cart/mini-cart pages, didn't tried from checkout page. The site redirects to paypal, then from paypal to magento again for order placement/review. On this page error appeared of "Wrong email address format".

But issue is resolved from the patch I have uploaded above. Express checkout works fine now.

**Comment by rommelfreddy** [comment]
well, i do not think this issue is resolved.
This issue ist resolved for you @Arsalanulhaq - but it seems so, that there is a bug in the paypal module.

@VigneshVenkatesan154 i will create the video the next days.

**Comment by Arsalanulhaq** [comment]
Yes, I also did some debugging, issue was in Paypal module, the solution I have provided is also seemed to be temporary because it just set customer email again if it is not found. Btw this solution was provided by Magento support. I was able to create a ticket to Magento Support somehow, and they provided this patch.

**Comment by sskharate** [comment]
For anyone to come across this issue, here is the quick code to copy. Thanks to @Arsalanulhaq 
``` 
/* set customer email if email is found empty - Unbale to place guest order Paypal checkout */
    if($this->_quote->getCustomerEmail() === null && $this->_quote->getBillingAddress()->getEmail() !== null) {
        $this->_quote->setCustomerEmail($this->_quote->getBillingAddress()->getEmail());
    }
```

**Comment by thampe** [comment]
Hello evyerone,

why is this issue closed. This can be clearly reproduced. I have done so in one of our shops. The fix seems ok.

Do you need an pull request?

Thanks!

**Comment by Arsalanulhaq** [comment]
> Hello everyone,
> 
> why is this issue closed. This can be clearly reproduced. I have done so in one of our shops. The fix seems ok.
> 
> Do you need an pull request?
> 
> Thanks!

@thampe I think the fix is temporary, as we are just setting the customer email again if it is not found, so there should be a proper fix in core paypal module where it is retrieving the customer email from db-quote table. Although this fix worked in my 2 different projects as well.

**Comment by thampe** [comment]
Hi,

there already seems to be a fix for this in https://github.com/magento/magento2/issues/25399#issuecomment-694599298
> The fix will be available with the upcoming 2.4.2 release.

So this issue can stay closed :).
